src/localstream.js

```diff
@@ -73,8 +73,40 @@ const localstream = (function() {
 	 * @description Flag indicating if the audio and video initialisation is in progress
 	 */
 	let initAudioVideoProgress = false;
+	/**
+	 * @description videoSources array indicating the avaliable video sources
+	 */
+	const videoSources=[];
+	/**
+	 * @description currentVideoSource index indicating the currently used video sources
+	 */
+	let currentVideoSource=0;
+	/**
+	 *@description An array containing audio and video listeners as functions. These listeners will be triggered in case of initiating video/audio by switching camera.
+	 */
+	const streamListenersAudioVideo=[];
 
 	/**
+	 * Call back method for enumerateDevices 
+	 * that is used to get the sources attached to the currently used device/browser
+	*/
+	function gotSources(sourceInfos) {
+		for (let i = 0; i < sourceInfos.length; i++) {
+			const sourceInfo = sourceInfos[i];
+			if (sourceInfo.kind === 'videoinput') {
+				videoSources.push(sourceInfo);
+			} 
+		}
+	}
+	/**
+	 * a method to reterive all of the video sources avaliable connected to the current device.
+	 */
+	function _getAllVideoSources(){
+		if (videoSources.length===0) {
```

Comment:
enumerateDevices is an synchronous method so _getAllVideoSources should use a Promise based approach.

`function _getAllVideoSources() {
	if(videoSources.length > 0) {
		return Promise.resolve(videoSources);
	}
	return navigator.mediaDevices.enumerateDevices()
		.then(sourceInfos => {
			videoSources.concat(sourceInfos.filter(sourceInfo => /^videoinput$/.test(sourceInfo.kind)));
			return videoSources;
		});
}`